src/org/sosy_lab/java_smt/solvers/princess/PrincessModel.java

```diff
@@ -281,35 +288,40 @@ private ITerm simplifyRational(ITerm pTerm) {
 
   @Override
   protected @Nullable IExpression evalImpl(IExpression expr) {
+    // TODO creating our own utility variables might eb unexpected from the user.
+    //   We might need to exclude such variables in models and formula traversal.
```

Comment:
This should be fine as Princess keeps symbols "per stack level". Once the level is popped at the end of this method the symbol is deleted and won't show up in models in the future.

Reply:
Yes, good catch. I would still prefer to not need to use any additional symbols.